nio/block/mixins/retry/retry.py

```diff
@@ -124,6 +119,7 @@ def execute_with_retry(self, execute_method, *args, stop_retry_event=None,
             Exception: The exception that execute_method raised when the
                 backoff strategy decided to stop retrying.
         """
+        self.setup_backoff_strategy()
```

Comment:
This won't work either. It still shares the backoff strategy with all calls in this block. When things are synchronous this is fine, but you could asynchronously call two execute with retry-s and observe that they would share the resource. One would actually, in essence, "reset" the backoff strategy while the other was still using it.

To make this work, you could have `setup_backoff_strategy` return an instance of a backoff strategy rather than save it to the instance of this class. Then use that return value throughout the `execute_with_retry` method.

Reply:
I added an assertion to the test that fails if Strategy is not a unique instance for each call of `execute_with_retry`, and it is, it really looks to me that I have two unique instances here:
https://github.com/niolabs/nio/pull/108/commits/cbf95eb0450b17730a7cd266af3bfa30a94188db#diff-b16c6262821a3775fca7f8702081d02aR89

What I have not been able to make a good test for is checking that the `retry_num` attribute is not shared between them, but based on the above I think that each call gets its own `retry_num`...